scripts/http-jsonp-detection.nse

```diff
@@ -0,0 +1,188 @@
+local http = require "http"
+local shortport = require "shortport"
+local stdnse = require "stdnse"
+local string = require "string"
+local json = require "json"
+local url = require "url"
+local httpspider = require "httpspider"
+local table = require "table"
+
+description = [[
+Attempts to discover JSONP endpoints in web servers. JSONP endpoints can be
+used to bypass Same-origin Policy restrictions in web browsers.
+
+The script searches for callback functions in the response to detect JSONP
+endpoints. It also tries to determine callback function through URL(callback 
+function may be fully or partially controllable from URL) and also tries to 
+bruteforce the most common callback variables through the URL.
+
+References : https://securitycafe.ro/2017/01/18/practical-jsonp-injection/
+
+]]
+
+---
+-- @usage
+-- nmap -p 80 --script http-jsonp-detection <target>
+--
+-- @output
+-- 80/tcp open  http    syn-ack
+-- | http-jsonp-detection: 
+-- | The following JSONP endpoints were detected: 
+-- |_/rest/contactsjp.php  Completely controllable from URL
+-- 
+--
+-- @xmloutput
+-- <table key='jsonp_endpoints'>
+-- <elem>/rest/contactsjp.php</elem>
+-- </table>
+-- 
+-- @args http-jsonp-detection.path The URL path to request. The default path is "/".
+--- 
+
+author = {"Vinamra Bhatia"}
+license = "Same as Nmap--See https://nmap.org/book/man-legal.html"
+categories = {"safe", "vuln", "discovery"}
+
+portrule = shortport.http
+
+local callbacks = {"cb", "jsonp", "jsonpcallback", "jcb", "call"}
+
+--Checks the body and returns if valid json data is present in callback function
+local checkjson = function(body)
+  
+  local func, json_data
+  _, _, func, json_data = string.find(body, "^([%w_]*)%((.*)%)$") 
```

Comment:
I think you want to replace here * for + as you want at least 1 match of this characters. * is for 0 or more and + is for at least 1.